test/src/kbasesearchengine/test/main/SearchMethodsTest.java

```diff
@@ -669,4 +673,58 @@ private void compare(final SortingRule got, final SortingRule expected) {
                 is(expected.getIsObjectProperty() == null ? 1L : expected.getIsObjectProperty()));
     }
 
+    @Test
+    public void testAccessFilter() throws Exception{
+        final AccessGroupProvider agp = mock(AccessGroupProvider.class);
+        final IndexingStorage is = mock(IndexingStorage.class);
+        final TypeStorage ts = mock(TypeStorage.class);
+
+        final Set<String> admins = new HashSet<>(Arrays.asList("auser"));
+        final SearchInterface sm = new SearchMethods(agp, is, ts, admins);
+
+        final List<Integer> adminList = new ArrayList<>();
+        adminList.add(2);
+        final Map<String, Integer> exMap = new HashMap<>();
+        exMap.put("test", 1);
+
+        when(agp.findAccessGroupIds("auser"))
+                .thenReturn(adminList);
+        kbasesearchengine.search.AccessFilter test  = new kbasesearchengine.search.AccessFilter()
+                .withAccessGroups(new LinkedHashSet(adminList));
+
+        //return data is user right access group id
+        when(is.searchTypes( kbasesearchengine.search.MatchFilter.getBuilder().build(),
+                new kbasesearchengine.search.AccessFilter()
+                        .withAccessGroups(new LinkedHashSet(adminList))
+                        .withAdmin(true)))
+        .thenReturn(exMap);
+
+
+        //expected private results: SearchMethods changes integer results to longs
+        final Map<String, Long> expectedRes = new HashMap<>();
+        expectedRes.put("test", 1L);
+
+        final SearchTypesInput input1 = new SearchTypesInput()
+                .withAccessFilter(new AccessFilter().withWithPrivate(1L))
+                .withMatchFilter(new MatchFilter());
+
+        //authorized and showing private data
+        final SearchTypesOutput res = sm.searchTypes(input1,
+                "auser");
+
+        assertThat("incorrect res", res.getTypeToCount(), is(expectedRes));
+
+        //unauthorized user
```

Comment:
These next two tests (which should be in separate tests) will always pass, AFAICT, because they depend on the default returns from Mockito. The only way they could fail is if they happened to trigger the stub set up above.

Reply:
so the last one is basically checking that the stub is not being triggered. If I add another stub for the no user case, that should be it?

Reply:
The cases I see are: 
authorized user and showing private data
authorized user and not showing private data
unauthorized user (private data is hard coded as false here)

Reply:
There's 5 cases I think:
Unauthorized (private = false, public = true) <- hardcoded booleans
Authorized with:
private = false, public = true
private = true, public = false
private = true, public = true
private = false, public = false <- this is the stupid case

Reply:
wait why am i testing public. didnt change anything with regards to that?

Reply:
public did change: https://github.com/kbase/KBaseSearchEngine/pull/260/files#diff-f58b37782d687c1180cb73e938fcec71R127